src/wp-includes/cron.php

```diff
@@ -155,7 +192,17 @@ function wp_schedule_single_event( $timestamp, $hook, $args = array() ) {
 		'args'     => $event->args,
 	);
 	uksort( $crons, 'strnatcasecmp' );
-	return _set_cron_array( $crons );
+
+	$set = _set_cron_array( $crons );
```

Comment:
The reason I didn't is because it's just a thin wrapper around `update_option()`. Now you mention it, it might be a good idea in case we can determine the reason for failure inside of it. I'll make the change.